src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-writer.c

```diff
@@ -2220,18 +2220,22 @@ write_sriov_setting (NMConnection *connection, shvarFile *ifcfg)
 	NMSriovVF *vf;
 	char key[32];
 	char *str;
+	gboolean enable = FALSE;
 
 	svUnsetAll (ifcfg, SV_KEY_TYPE_SRIOV_VF);
 
 	s_sriov = NM_SETTING_SRIOV (nm_connection_get_setting (connection, NM_TYPE_SETTING_SRIOV));
 	if (s_sriov)
-		num = nm_setting_sriov_get_total_vfs (s_sriov);
-	if (num == 0) {
+		enable = nm_setting_sriov_get_enable (s_sriov);
+	if (!enable) {
```

Comment:
the reader/writer should be able to persist a setting without loss (full round-trip). Ok, SRIOV's `verify()` enforces that if the setting is not enabled, that the setting isn't present (somewhat). So, currently it doesn't make a difference. But in general, the ifcfg code should not have additional conditions on top of what NMSettingSriov already has. In the best case, it checks impossible conditions, in the worst case, the writer does not persist every possible setup. Less logic here, just plain setting the fields.

Reply:
This should be fixed now.